scripts/halium

```diff
@@ -466,6 +466,12 @@ mountroot() {
 		halium_panic "Couldn't find data partition."
 	fi
 
+	# Format a wiped userdata without a filesystem header
+	if head -c 4096 $path | cmp /dev/zero 2>&1 | grep -q '^cmp: EOF on -$'; then
```

Comment:
My concern is probably how reliable our detection is i.e. what's the likelihood we format over some data.

IMO I think 4KiB is probably enough to tell if a disk is wiped or not, and the method is probably sound. Though my feeling says that there's probably a better way to do this.

[This answer](https://unix.stackexchange.com/a/713109) from StackExchange suggests using a command called `badblock` (part of `e2fsprogs) instead; I'm not sure if it's better than this method or not.

Reply:
To clarify, that reflects how Android does it: https://github.com/LineageOS/android_system_core/blob/lineage-22.1/libcutils/partition_utils.cpp#L42.

Reply:
Ok then.